crates/extrinsics/src/lib.rs

```diff
@@ -347,29 +384,58 @@ pub async fn fetch_contract_info(
     contract: &AccountId32,
     rpc: &LegacyRpcMethods<DefaultConfig>,
     client: &Client,
-) -> Result<Option<ContractInfo>> {
-    let info_contract_call = api::storage().contracts().contract_info_of(contract);
-
+) -> Result<ContractInfo> {
     let best_block = get_best_block(rpc).await?;
 
-    let contract_info_of = client
+    let contract_info_address = dynamic(
+        "Contracts",
+        "ContractInfoOf",
+        vec![Value::from_bytes(contract)],
+    );
+    let contract_info = client
         .storage()
         .at(best_block)
-        .fetch(&info_contract_call)
-        .await?;
+        .fetch(&contract_info_address)
+        .await?
+        .ok_or_else(|| {
+            anyhow!(
+                "No contract information was found for account id {}",
+                contract
+            )
+        })?;
+    #[derive(scale_decode::DecodeAsType, Debug)]
+    pub struct BoundedVec<T>(pub ::std::vec::Vec<T>);
+    #[derive(scale_decode::DecodeAsType, Debug)]
+    struct ContractInfoOf {
+        trie_id: BoundedVec<u8>,
+        code_hash: CodeHash,
+        storage_items: u32,
+        storage_item_deposit: Balance,
+    }
+    #[derive(scale_decode::DecodeAsType)]
+    struct DepositAccount {
+        deposit_account: AccountId32,
+    }
 
-    match contract_info_of {
-        Some(info_result) => {
-            let convert_trie_id = hex::encode(info_result.trie_id.0);
-            Ok(Some(ContractInfo {
-                trie_id: convert_trie_id,
-                code_hash: info_result.code_hash,
-                storage_items: info_result.storage_items,
-                storage_item_deposit: info_result.storage_item_deposit,
-            }))
+    let total_balance: Balance = match contract_info.as_type::<DepositAccount>() {
+        Ok(account) => {
+            // StorageVersion >= 10 and < 15 contains deposit in separate account
+            let deposit_account = AccountId32(account.deposit_account.0);
```

Comment:
```suggestion
            let deposit_account = &account.deposit_account;
```
Don't need to construct a new `AccountId32`?